dll/win32/browseui/shellbrowser.cpp

```diff
@@ -2257,12 +2311,35 @@ HRESULT STDMETHODCALLTYPE CShellBrowser::TranslateAcceleratorSB(MSG *pmsg, WORD
 
 HRESULT STDMETHODCALLTYPE CShellBrowser::BrowseObject(LPCITEMIDLIST pidl, UINT wFlags)
 {
-    if ((wFlags & SBSP_EXPLOREMODE) != NULL)
+    UINT newbrowser = GetNewBrowserFlag(wFlags);
+    // FIXME: Should not automatically show the Explorer band
+    if ((wFlags & SBSP_EXPLOREMODE) && !(newbrowser & SBSP_NEWBROWSER))
         ShowBand(CLSID_ExplorerBand, true);
 
+    CComHeapPtr<ITEMIDLIST> pidlResolved;
+    if (wFlags & (SBSP_RELATIVE | SBSP_PARENT))
+    {
+        HRESULT hr = CreateRelativeBrowsePIDL(pidl, wFlags, &pidlResolved);
+        if (FAILED(hr))
+            return hr;
+        pidl = pidlResolved;
+    }
+
+    if (newbrowser & SBSP_NEWBROWSER)
+        return OpenNewBrowserWindow(pidl, wFlags);
+    if (wFlags & SBSP_PARENT)
+        return NavigateToParent();
+    if (wFlags & SBSP_NAVIGATEBACK)
+        return GoBack();
+    if (wFlags & SBSP_NAVIGATEFORWARD)
```

Comment:
I wonder what happens on Windows if both `SBSP_NAVIGATEBACK` and `SBSP_NAVIGATEFORWARD` are set at the same time in `wFlags` :thinking:

Reply:
Probably undefined behavior. The same applies to SBSP_PARENT.